### PR TITLE
Dragonfly signature lookups

### PR DESCRIPTION
**Pull request by lbiedinger**

### CODE REVIEW

File: app/jobs/europeana/picture_versions_job.rb [hunk 1]
```diff
@@ -4,10 +4,13 @@ class Europeana::PictureVersionsJob < ActiveJob::Base
 
   def perform(id)
     picture = Alchemy::Picture.find(id)
-    Europeana::Elements::Image::VERSIONS.values.compact.each do |settings|
+    Europeana::Elements::Image::VERSIONS.each_pair do |version_key, settings|
       Rails.logger.info "Creating #{JSON.generate(settings)} of picture #{id}"
       next if settings[:size].nil?
-      picture_version(picture, settings).data
+      version = picture_version(picture, settings)
+      # call .data on the version to ensure it is persisted
```
Comment (rwd) [inline app/jobs/europeana/picture_versions_job.rb]: What does this mean?

File: app/models/alchemy/dragonfly_signature.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class DragonflySignature < ActiveRecord::Base
```
Comment (rwd) [inline app/models/alchemy/dragonfly_signature.rb]: Could we have ActiveModel validations for the presence and uniqueness constraints enforced by the schema?

File: db/migrate/20171207152149_create_alchemy_dragonfly_signatures.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class CreateAlchemyDragonflySignatures < ActiveRecord::Migration
+  def up
```
Comment (rwd) [inline db/migrate/20171207152149_create_alchemy_dragonfly_signatures.rb]: Replace `#up` and `#down` with `#change`

File: db/migrate/20171207152149_create_alchemy_dragonfly_signatures.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    return if table_exists?(:alchemy_dragonfly_signatures)
```
Comment (rwd) [inline db/migrate/20171207152149_create_alchemy_dragonfly_signatures.rb]: This should not be necessary.

File: lib/tasks/jobs.rake [hunk 5]
```diff
@@ -9,4 +9,13 @@ namespace :jobs do
       end
     end
   end
+
+  namespace :pictures do
+    desc 'Enqueue picture versions jobs to generate Alchemy::PictureVersions'
+    task version_jobs: :environment do
+      Alchemy::Picture.all.each do |picture|
```
Comment (rwd) [inline lib/tasks/jobs.rake]: This should use `.find_each` instead of `.all.each`

File: lib/europeana/mixins/image_version.rb [hunk 6]
```diff
@@ -37,7 +37,8 @@ def versions(name = 'image')
           if image&.essence&.picture.present?
             VERSIONS.each_with_object({}) do |(version, settings), memo|
```
Comment (rwd) [inline lib/europeana/mixins/image_version.rb]: In the definition of `VERSIONS`, why are the `quality` keys sometimes just `quality`, somtimes `'quality'`?

Comment (lbiedinger) [reply lib/europeana/mixins/image_version.rb]: no idea, I'll update them to be consistent though.

File: app/models/alchemy/dragonfly_signature.rb (hunk 2 above)
Comment (lbiedinger) [reply app/models/alchemy/dragonfly_signature.rb]: initially I had not included these on purpose to keep this model as light as possible.
I've updated it to at least cascade delete the DragonflySignatures when image versions are deleted.

**Pull request merged**